source/PlayerInfo.cpp

```diff
@@ -3477,6 +3477,19 @@ void PlayerInfo::RegisterDerivedConditions()
 		return retVal;
 	});
 
+	// Check if an attribute is present in the player's fleet.
+	auto &&shipAttributeProvider = conditions.GetProviderPrefixed("ship attribute: ");
+	auto shipAttributeFun = [this](const string &name) -> int64_t
+	{
+		string attrib = name.substr(strlen("ship attribute: "));
+		int64_t retVal = 0;
+		for(const shared_ptr<Ship> &ship : ships)
+			retVal += ship->Attributes().Get(attrib) * 1000.;
```

Comment:
This should be making use of the `flagshipAttributeHelper` function (which could be renamed to just `shipAttributeHelper`). I also think that the search space of this condition should match the search space of the other "default" conditions that check your ships. That is, `"ship attribute: <attr>"` should only look for ships which are not parked or disabled and in your system, whereas `"ship attribute (all): <attr>"` would check everything that isn't destroyed. (See `"ships: <category>"` and `"ship model: <model>"` just above this for examples.)